pre_commit_hooks/check_executables_have_shebangs.py

```diff
@@ -8,23 +8,36 @@
 from typing import NamedTuple
 from typing import Sequence
 
+from pre_commit_hooks.util import CalledProcessError
 from pre_commit_hooks.util import cmd_output
 from pre_commit_hooks.util import zsplit
 
 EXECUTABLE_VALUES = frozenset(('1', '3', '5', '7'))
 
 
 def check_executables(paths: list[str]) -> int:
-    if sys.platform == 'win32':  # pragma: win32 cover
-        return _check_git_filemode(paths)
-    else:  # pragma: win32 no cover
-        retv = 0
-        for path in paths:
-            if not has_shebang(path):
-                _message(path)
-                retv = 1
-
-        return retv
+    fs_tracks_executable_bit = cmd_output(
+        'git', 'config', 'core.fileMode', retcode=None,
+    )
+    if fs_tracks_executable_bit != 'true':  # pragma: win32 cover
```

Comment:
this condition is always false so your entire testsuite and changes in this and the other PR are incorrect